internal/service/project/resource_project_test.go

```diff
@@ -167,43 +167,38 @@ func TestUpdateProject(t *testing.T) {
 			name:         "Successful update",
 			projectState: projectStateName,
 			projectPlan:  projectStateNameDiff,
-			mockResponses: []ProjectResponse{
-				{
-					Err: nil,
-				},
+			mockResponses: ProjectResponse{
+				Err: nil,
 			},
 			expectedError: false,
 		},
 		{
 			name:         "Same project names; No update",
 			projectState: projectStateName,
 			projectPlan:  projectStateName,
-			mockResponses: []ProjectResponse{
-				{
-					Err: nil,
-				},
+			mockResponses: ProjectResponse{
+				Err: nil,
 			},
 			expectedError: false,
 		},
 		{
 			name:         "Failed API call; Failed update",
 			projectState: projectStateName,
 			projectPlan:  projectStateNameDiff,
-			mockResponses: []ProjectResponse{
-				{
-					ProjectResp:  nil,
-					HTTPResponse: &http.Response{StatusCode: 503},
-					Err:          errors.New("Service Unavailable"),
-				},
+			mockResponses: ProjectResponse{
+				ProjectResp:  nil,
+				HTTPResponse: &http.Response{StatusCode: 503},
+				Err:          errors.New("Service Unavailable"),
 			},
 			expectedError: true,
 		},
 	}
 
 	for i, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			testObject := new(MockProjectService)
-			testObject.On("UpdateProject", mock.Anything, mock.Anything).Return(tc.mockResponses[0])
+			testObject := mocksvc.NewGroupProjectService(t)
+
+			testObject.On("UpdateProject", mock.Anything, mock.Anything, mock.Anything).Return(tc.mockResponses.ProjectResp, tc.mockResponses.HTTPResponse, tc.mockResponses.Err).Maybe()
```

Comment:
some test cases will not call the mocked method, so we add the `.Maybe()` "Maybe allows the method call to be optional. Not calling an optional method will not cause an error while asserting expectations". This is needed because when calling `NewGroupProjectService`, a cleanup function is registered to assert the mocks expectations

Reply:
is there any method that we call always and doesn't need maybe?

Reply:
there was only one, removed the maybe.